src/Catel.MVVM/Catel.MVVM.Shared/Collections/FastObservableCollection.cs

```diff
@@ -422,6 +443,46 @@ public IDisposable SuspendChangeNotifications(SuspensionMode mode)
         }
 
         /// <summary>
+        /// Synchroniz the list with the suspenstion context.
+        /// </summary>
+        private void SynchronizeFromSuspensionContext()
+        {
+            if (_suspensionContext.NewItems.Count > 0)
+            {
+                _suspensionContext.NewItemIndices.Reverse();
+                _suspensionContext.NewItems.Reverse();
+
+                for (int i = 0; i < _suspensionContext.NewItems.Count; i ++)
+                {
+                    base.InsertItem(_suspensionContext.NewItemIndices[i], _suspensionContext.NewItems[i]);
+                }
+            }
+
+            if (_suspensionContext.OldItems.Count > 0)
+            {
+                SortedList<int, T> sortedList = new SortedList<int, T>();
+                for (int i = 0; i < _suspensionContext.OldItems.Count; i++)
+                {
+                    var value = _suspensionContext.OldItems[i];
+                    var idx = _suspensionContext.OldItemIndices[i];
+                    sortedList.Add(idx, value);
```

Comment:
SortedList.Add has a complexity of O(n), so this will be O(n^2).
Replace it at least with SortedDictionary as it has O(log n) for adding resulting in O(n log n).